### PR TITLE
[MAINTENANCE] Add support for PHP 8.3.x

### PR DESCRIPTION
**Pull request by stweil**
composer.lock was updated with `composer update`.

This also fixes a Dependabot alert.

### CODE REVIEW

**Comment by stweil** [comment]
It looks like this change requires more work. CI fails partially.

**Comment by sebastian-meyer** [comment]
All tests run with PHP 7.4 because that's the newest version TYPO3 v10 supports. Therefore we can't use dependencies which require newer PHP versions.
You have to run `composer update` with PHP 7.4 in order to get the latest possible dependencies.

**Comment by stweil** [comment]
PHP 8.3 requires a newer version of Solarium. As far as I see that newer version is also available for PHP 7.4, but requires updates in the Kitodo code. As we want support for TYPO3 v11, these updates are mandatory.

**Comment by sebastian-meyer** [comment]
I don't think, updating Solarium is strictly mandatory since TYPO3v11 still officially supports PHP 7.4.

I know that PHP 7.4 is no longer supported itself, but I'd rather first publish Kitodo.Presentation 5.0 and then work on updating Solarium and supporting newer PHP versions up to 8.3 instead of further postponing the release and don't have a TYPO3v11 compatible version at all.

So my suggestion is releasing Kitodo.Presentation 5.0 which supports TYPO3v10 and v11, but only PHP 7.4, and then releasing Kitodo.Presentation 5.1 which supports TYPO3v10 (with PHP 7.4) and v11 up to PHP 8.3.

**Comment by stweil** [comment]
Sure, there are limited resources, so the open tasks must be done step by step.

But users must be aware of those steps, and the requirements which are listed in the README currently say that PHP 8.3.x is supported.

I'll continue to work on support for PHP 8.3 because that is important for us, so maybe we will be able to release a 5.0 which works for all combinations listed in the README.

**Comment by sebastian-meyer** [comment]
Thank you! I'll create a separate pull request just updating the dependencies for PHP 7.4 (and fixing the misleading requirements in the README) in order to get rid of the security issue at hand.

**Comment by stweil** [comment]
Meanwhile most tests pass with TYPO3 v11 and Solarium 6.0.4:
```
$ composer test -- -p 8.2
> Build/Test/runTests.sh -s unit '-p' '8.2'
Creating network "dlf_testing_default" with the default driver
Creating dlf_testing_unit_run ... done
PHP 8.2.12 (cli) (built: Oct 28 2023 02:10:03) (NTS)
PHPUnit 9.6.16 by Sebastian Bergmann and contributors.

..............S                                                   15 / 15 (100%)

Time: 00:00.018, Memory: 8.00 MB

OK, but incomplete, skipped, or risky tests!
Tests: 15, Assertions: 71, Skipped: 1.
Removing dlf_testing_unit_run_17c1c9b9bcf9 ... done
Removing network dlf_testing_default

###########################################################################
Result of unit
PHP: 8.2
SUCCESS
###########################################################################

> Build/Test/runTests.sh -s functional '-p' '8.2'
Using driver: mysqli
Creating network "dlf_testing_default" with the default driver
Creating dlf_testing_mariadb_1 ... done
Creating dlf_testing_web_1     ... done
Creating dlf_testing_solr_1    ... done
Creating dlf_testing_functional_run ... done
Waiting for database start...
Database is up
Waiting for Solr start...
Solr is up
PHP 8.2.12 (cli) (built: Oct 28 2023 02:10:03) (NTS)
PHPUnit 9.6.16 by Sebastian Bergmann and contributors.

.....F..................................................          56 / 56 (100%)

Time: 00:18.692, Memory: 207.00 MB

There was 1 failure:

1) Kitodo\Dlf\Tests\Functional\Api\OaiPmhTest::noRecordsUntil1900
Failed asserting that exception of type "Phpoaipmh\Exception\MalformedResponseException" matches expected exception "Phpoaipmh\Exception\OaipmhException". Message was: "Could not decode XML Response: String could not be parsed as XML" at
/kitodo-presentation/vendor/caseyamcl/phpoaipmh/src/Client.php:134
/kitodo-presentation/vendor/caseyamcl/phpoaipmh/src/Client.php:95
/kitodo-presentation/vendor/caseyamcl/phpoaipmh/src/RecordIterator.php:224
/kitodo-presentation/vendor/caseyamcl/phpoaipmh/src/RecordIterator.php:190
/kitodo-presentation/vendor/caseyamcl/phpoaipmh/src/RecordIterator.php:326
/kitodo-presentation/Tests/Functional/Api/OaiPmhTest.php:179
.

FAILURES!
Tests: 56, Assertions: 263, Failures: 1.
ERROR: 1
Stopping dlf_testing_solr_1    ... done
Stopping dlf_testing_web_1     ... done
Stopping dlf_testing_mariadb_1 ... done
Removing dlf_testing_functional_run_aa9bbc6415bd ... done
Removing dlf_testing_solr_1                      ... done
Removing dlf_testing_web_1                       ... done
Removing dlf_testing_mariadb_1                   ... done
Removing network dlf_testing_default

###########################################################################
Result of functional
PHP: 8.2
DBMS: mariadb  version 10.3  driver mysqli
FAILURE
###########################################################################

Script Build/Test/runTests.sh -s functional handling the test:func event returned with error code 1
Script @test:func was called via test
```

**Comment by stweil** [comment]
@sebastian-meyer, it looks like the Unit and Functional Testing installed Solarium 6.0.4, and all tests passed. Maybe we can drop support for Solarium 5. Removing that code would also fix the warning from PHPStan.

File: composer.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "typo3/cms-tstemplate": "^10.4.37|^11.5.33",
     "caseyamcl/phpoaipmh": "^3.3.1",
     "ubl/php-iiif-prezi-reader": "0.3.0",
-    "solarium/solarium": "^5.2.0"
+    "solarium/solarium": "5.2.0 - 6.0.99"
```
Comment (stweil) [inline composer.json]:
```suggestion
    "solarium/solarium": "5.2.0 - 6.3"
```

Comment (stweil) [reply composer.json]: Solarium 6.3.5 seems to work as good as 6.0.x (most CI tests pass, one error for OAI).

**Comment by stweil** [comment]
@sebastian-meyer, the last commit is only indirectly related to the rest: I have to use a different TCP port for the test because port 8000 is not free on my test system. If desired, I can remove it here and write a separate pull request for it.

**Comment by stweil** [comment]
@sebastian-meyer, the pull request is now ready for merging. If you prefer to stick to Solarium 5 for now, the first commit can be skipped. I think that the remaining four commits are safe for the upcoming release.

**Comment by stweil** [comment]
Still missing: as soon as all commits were merged, the documentation (at least README.md) needs an update, too, to show that PHP 8.3 is now supported.

**Comment by stweil** [comment]
@sebastian-meyer, I think this PR could also be included in the planned release.

**Review by sebastian-meyer — APPROVED** [review]

**Pull request merged**